src/services/market.service.ts

```diff
@@ -86,13 +89,42 @@ export class MarketService {
         return val.market_cap;
     };
 
-    getTokensPrices = async (symbols: string[], forceRefresh = false): Promise<{ symbol: string; price: number }[]> => {
-        if (!this._informations.length || forceRefresh) {
-            await this.refreshTokenInformations();
-        }
+    saveBulk = async (data: MarketData[]): Promise<MarketEntity> => {
```

Comment:
Method is improperly named. It does not save bulk market entities. It takes a bulk of market data, then construct a new market entity and save it.